sshproxy.go

```diff
@@ -166,73 +167,89 @@ func (s *Server) handleRequests(reqs <-chan *ssh.Request, channel ssh.Channel, c
 				// append client env to the command environment
 				keyLen := req.Payload[3]
 				valLen := req.Payload[keyLen+7]
-				key := string(req.Payload[4 : keyLen+4])
-				val := string(req.Payload[keyLen+8 : keyLen+valLen+8])
+				key := string(req.Payload[4: keyLen+4])
+				val := string(req.Payload[keyLen+8: keyLen+valLen+8])
 				env = append(env, fmt.Sprintf("%s=%s", key, val))
 			}
 			req.Reply(s.passEnv, nil)
 		case "pty-req":
-			// client has requested a PTY
-			wantsPty = true
-			req.Reply(true, nil)
-		case "shell":
-			// client has not supplied a command, reject
-			req.Reply(false, nil)
+			ok := terminal == nil
+			if ok {
+				var err error
+				terminal, err = pty.NewTerminal()
+				ok = err == nil
```

Comment:
This wasn't supposed to make it into the final commit, at the time I didn't care what the error was but I wanted to communicate back to the client if the terminal had been created.